### PR TITLE
Updated PhpDoc, mainly made with Psalm

### PR DESCRIPTION
**Pull request by HenkPoley**
Patch for https://github.com/FreeDSx/LDAP/issues/34

Can't say that I'm happy that it's such a big patch. But at least I managed to keep it to just the doctype with grepdiff.

### CODE REVIEW

File: src/FreeDSx/Ldap/Control/Ad/DirSyncRequestControl.php [hunk 1]
```diff
@@ -143,7 +143,11 @@ public function setCookie(string $cookie)
     }
 
     /**
-     * {@inheritdoc}
+     * @param AbstractType $type
+     * @return Control
+     * @throws ProtocolException
+     * @throws \FreeDSx\Asn1\Exception\EncoderException
+     * @throws \FreeDSx\Asn1\Exception\PartialPduException
```
Comment (ChadSikorra) [inline src/FreeDSx/Ldap/Control/Ad/DirSyncRequestControl.php:149]: Was this an explicit change, or just how Psalm decided to do it? I think updating the interface this is coming from -- `ProtocolElementInterface` -- might be better. As in just adding additional throws to it. Helps avoid some duplicated docblock anyway. Though I can see why it potentially did this if it wants to always return the specific type in the docblock and not the abstract type from the interface

Comment (HenkPoley) [reply src/FreeDSx/Ldap/Control/Ad/DirSyncRequestControl.php:150]: It is just from Psalm telling that these Exceptions can be seen there.

This should be pre-existing confusing behaviour. Just previously undocumented.

Comment (HenkPoley) [reply src/FreeDSx/Ldap/Control/Ad/DirSyncRequestControl.php:149]: Ah, these `@inheritdoc` replacements. Psalm is barely aware of the interaction between Interfaces/Traits and classes.

But sure. I can review them, though it is kind of tedious to review, and currently still correct.

**Comment by HenkPoley** [comment]
My angle is that I just want all the splat (`...`) and exceptions (`@throws`) correctly documented. And whether it looks pretty or minimal in the docblocks, I don't really care that much. But I have some changes half-ready to fix the remark about `{@inheritDoc}`.

**Comment by ChadSikorra** [comment]
> My angle is that I just want all the splat (`...`) and exceptions (`@throws`) correctly documented. And whether it looks pretty or minimal in the docblocks, I don't really care that much. But I have some changes half-ready to fix the remark about `{@inheritDoc}`.

I like the changes for the varaidic methods, exceptions, and docblock corrections! I'd just like to keep it somewhat clean and avoid duplication if possible. If it ends up being too much effort I might just accept this as-is. I can always run CS fixer with some other rules to clean up some things.

I appreciate the work you've put into it.

**Comment by HenkPoley** [comment]
Fixing the things that phpstan is complaining about. But at least 1 is a bug in phpstan: https://github.com/phpstan/phpstan/issues/4843#issue-856122224

Not quite sure if the `?? 0` "fixes" are appropriate.

**Comment by ChadSikorra** [comment]
This seems good enough to me. It does expose lots of exceptions that I'm not sure I wanted library users to care about (as they would be quite rare to receive in a real setup, but still possible). Though overall I think this is good. Thanks for the work on it!

**Pull request merged**